lib/oli/resources/alternatives.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Oli.Resources.Alternatives do
+  alias Oli.Resources.Alternatives.AlternativesStrategyContext
+  alias Oli.Resources.Alternatives.SelectAllStrategy
+  alias Oli.Resources.Alternatives.UserSectionPreferenceStrategy
+
+  @doc """
+  Selects one or more alternatives using the element's specified strategy.
+
+  Returns a list of `Oli.Resources.Alternatives.Selection` structs.
+  """
+  def select(
```

Comment:
Determines which strategy to use based on the element's "strategy" field. Really after this PR the only strategy that is possible to create in the editor is "user_section_preference"